src/Microsoft.AspNet.Mvc.Core/Rendering/Html/HtmlHelper.cs

```diff
@@ -534,19 +534,19 @@ public static string GetFormMethodString(FormMethod method)
         /// <inheritdoc />
         public HtmlString TextArea(string expression, string value, int rows, int columns, object htmlAttributes)
         {
-            var metadata = ExpressionMetadataProvider.FromStringExpression(expression, ViewData, MetadataProvider);
+            var modelExplorer = ExpressionMetadataProvider.FromStringExpression(expression, ViewData, MetadataProvider);
             if (value != null)
             {
-                metadata.Model = value;
+                modelExplorer = modelExplorer.GetExplorerForExpression(typeof(string), value);
```

Comment:
it's expected that `value` will contain a formatted version of something with a non-`string` type.  `MultilineTextTemplate()` goes further and `ToString()`s the value when calling `TextArea()`.  problems arise if the real metadata includes for example validation attributes.
three choices:
1. plumb the requested value through to `DefaultHtmlGenerator.GenerateTextArea()` -- a bit like how `GenerateTextBox()` works.  I lean toward this choice.
2. add a `string ValueString` or perhaps `object ValueOverride` property to `RuntimeModelMetadata` (nee `ModelExplorer`)
3. allow `RuntimeModelMetadata.ModelMetadata` and `Model` to be incompatible.  this is the safest option, if we expect this method is one of a few hidden "display this representation" cases.